### ISSUE TITLE
Edit the twisted.internet docstring to make contents clearer

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2026]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2026|
|Type|enhancement|
|Created|2006-08-29 07:02:33Z|
|Branch|https://github.com/twisted/twisted/tree/improve-docstring-2026|

The contents of twisted.internet are unclear from the docstring. It definitely contains non-internet things, like defer and serialport.

<details><summary>Searchable metadata</summary>

```
trac-id__2026 2026
type__enhancement enhancement
reporter__jml jml
priority__normal normal
milestone__ 
branch__branches_improve_docstring_2026 branches/improve-docstring-2026
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1156834953000000 1156834953000000
changetime__1224793732000000 1224793732000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [24895]) Branching to 'improve-docstring-2026'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

I added a description, based on the [internet-overview](http://twistedmatrix.com/projects/core/documentation/howto/internet-overview.html) document.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

`twisted.internet` no longer contains a powerful persistence API (if it ever did).  Please remove that part of the docstring and remove that part of the internet overview. :)

I also find _compatible collection of event-loops_ to be somewhat strange.  _collection of compatible event-loops_ makes much more sense to me.

As long as you're editing the internet overview, can you make the interfaces it mentions into API links?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Made the changes in r25030, putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Can you add `IReactorMulticast` to the list in `internet-overview.xhtml`?  The comma in the middle of the sentence _It contains the code to dispatch events to interested observers, and a portable API so that observers need not care about which event loop is running._ should be removed (in the `twisted.internet` docstring and in `internet-overview.xhtml`).  The rest looks good, please merge when these two things are fixed.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [25064]) Merge improve-docstring-2026: Update the twisted.internet docstring to make the contents clearer. Remove outdated info from internet overview document. 

Author: thijs
Reviewer: exarkun
Fixes: [#2026](https://github.com/twisted/twisted/issues/2026)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
